core/src/kura.rs

```diff
@@ -328,6 +347,35 @@ impl Kura {
         data.pop();
         data.push((block.hash(), Some(block)));
     }
+
+    /// Store WSV to disk.
+    pub fn store_wsv(&self, wsv: &WorldStateView) {
+        let data: WSVData = wsv.into();
+        let path = self.block_store_path.join(WSV_FILE_NAME);
+
+        std::thread::spawn(move || {
+            let bin_data: Vec<u8> = data.encode();
+
+            let mut wsv_file = std::fs::OpenOptions::new()
+                .write(true)
+                .truncate(true)
+                .create(true)
+                .open(path.clone())
+                .expect("Failed to open wsv file");
+
+            wsv_file
+                .write_all(&bin_data)
+                .expect("Failed to write wsv file");
+        });
+    }
+
+    /// Load WSVData from disk.
+    pub fn try_load_wsv(&self) -> Option<WSVData> {
```

Comment:
To be honest what Kura looks now may lead to *Zombie* state. This is the problem known to lazy initialization and etc.

So user can create Kura with `new()` and may forget to load `wsv`. I understand that with this approach you're trying to solve different loading modes problem. What I suggest is to have one creation method which will accept loading mode.